tests/acceptance/Settings/SettingsArchivePageCest.php

```diff
@@ -51,7 +53,9 @@ public function createArchivePageWithVariousStatusNewsletters(\AcceptanceTester
     $i->waitForText('SentNewsletter');
     $i->dontSee('DraftNewsletter');
     $i->dontSee('ScheduledNewsletter');
+    $newsletterFactory = new Newsletter();
     $newsletterFactory->withSubject('SentNewsletter2')->withDraftStatus()->withSendingQueue()->withSegments([$segment3])->create();
+    $newsletterFactory = new Newsletter();
```

Comment:
This is a similar problem as at the Subscriber. The factory shouldn't be created more times. The method `withSegments` adds the segment again to the array. The solution can be to store the segment with a key which can be segment ID for example.